R/interactions.R

```diff
@@ -259,7 +259,7 @@ make_new_formula <- function(x) {
 ## term expansion (without `.`s). This returns the factor
 ## names and would not expand dummy variables.
 get_term_names <- function(form, vnames) {
-  if (!is_formula(form)) {
+  if (!rlang::is_formula(form, scoped = TRUE)) {
```

Comment:
When using `step_interactions()` the formula gets passed around as a language object instead of a formula. `recipes:::is_formula()` didn't care, `rlang::is_formula()` cares hence `scoped = TRUE`.